extensions/aws/s3/S3ClientRequestSender.cpp

```diff
@@ -31,8 +31,8 @@ namespace s3 {
 std::optional<Aws::S3::Model::PutObjectResult> S3ClientRequestSender::sendPutObjectRequest(
     const Aws::S3::Model::PutObjectRequest& request,
     const Aws::Auth::AWSCredentials& credentials,
-    const Aws::Client::ClientConfiguration& client_config) {
-  Aws::S3::S3Client s3_client(credentials, client_config);
+    gsl::not_null<Aws::Client::ClientConfiguration*> client_config) {
+  Aws::S3::S3Client s3_client(credentials, *client_config);
```

Comment:
Consider using a reference instead of `gsl::not_null<Aws::Client::ClientConfiguration*>`